src/main/java/com/hotels/housekeeping/service/impl/FileSystemHousekeepingService.java

```diff
@@ -47,39 +48,65 @@ public FileSystemHousekeepingService(LegacyReplicaPathRepository legacyReplicaPa
     LOG.warn("{}.fixIncompleteRecord(LegacyReplicaPath) should be removed in future.", getClass());
   }
 
+  private FileSystem fileSystemForPath(LegacyReplicaPath cleanUpPath) {
+    LOG.info("Attempting to delete path '{}' from file system", cleanUpPath);
+    Path path = new Path(cleanUpPath.getPath());
```

Comment:
This method could take the Path type directly so you don't have to create it twice (once here and once in the calling method currently)

Reply:
Done